### ISSUE TITLE
# in URL breaks pattern matching in Reverse Mode

### ISSUE THREAD

**Issue by dyspr0sium** [andy-portmen/open-in#66]
In the Chrome version of Open In Firefox, I have the following set for the list of domains:
```
meet.google.com, chat.google.com
```
with Reverse Mode ticked. This should mean that any links matching those two domains should open in Chrome, and anything else should open in Firefox.

This works well, opening all other links in Firefox as intended, but if there is a `#` in the link, then the link incorrectly opens in Chrome.

E.g. 
[https://docs.python.org/3.8/library/typing.html](https://docs.python.org/3.8/library/typing.html) -> opens correctly in Firefox
[https://docs.python.org/3.8/library/typing.html#typing.Literal](https://docs.python.org/3.8/library/typing.html#typing.Literal) -> opens incorrectly in Chrome

[http://kieranwynn.github.io/pyquaternion](http://kieranwynn.github.io/pyquaternion) -> opens correctly in Firefox
[http://kieranwynn.github.io/pyquaternion/#object-initialisation](http://kieranwynn.github.io/pyquaternion/#object-initialisation) -> opens incorrectly in Chrome

This issue only occurs in Reverse Mode; in normal mode URLs in the list of domain with or without `#` are correctly opened in Firefox.

**Comment by dyspr0sium** [comment]
Looks like this behaviour might be caused by [line 34 in `open-in/common/data/inject.js`](https://github.com/andy-portmen/open-in/blob/0e675573baa95be3ea9999a1ea1a6f9e490e3a20/common/data/inject.js#L34):
```
if (config.reverse && a.href && (a.href.indexOf('#') === -1 || isTop)) {
    if (a.href.startsWith('http') || a.href.startsWith('file')) {
      return callback(a.href);
    }
  }
```
In reverse mode, the code is explicitly checking that `#` is not in the link. I'm sure there was a good reason to do so, but I'm not sure what it was and whether it can be differentiated from the cases in my examples where you would want the links to work normally.

**Comment by dyspr0sium** [comment]
I can confirm that this issue is solved on Open In Firefox 0.4.1. Thanks for the quick fix Andy.
